orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/graph/StageGraphBuilder.java

```diff
@@ -148,11 +148,18 @@ public void connect(@Nonnull Stage previous, @Nonnull Stage next) {
   }
 
   private String generateRefId() {
+    long offset = parent
```

Comment:
If we're adding stages to an already built execution, it's important to not re-use a `refId`.